src/lib/sql/sql_pipeline_statement.cpp

```diff
@@ -146,6 +146,14 @@ const std::shared_ptr<AbstractLQPNode>& SQLPipelineStatement::get_optimized_logi
     return _optimized_logical_plan;
   }
 
+  // Handle logical query plan if statement has been cached
+  if (const auto cached_plan = SQLQueryCache<std::shared_ptr<AbstractLQPNode>>::get().try_get(_sql_string)) {
```

Comment:
Using something called "SQLQueryCache" to cache LQPs. Ouch, not your fault, that thing should probably just be called "LockedCache" or something

Reply:
Noted. (But this seems out of scope for this PR)